### PR TITLE
[12.0][ADD] New module purchase product weight

### PR DESCRIPTION
**Pull request by sergiocorato**
This module ease the use of product managed on any unit of measure but purchased on weight, without adding multiple conversion, but simply using weight on product. A future extension can be use of volume too.

### CODE REVIEW

**Comment by sergiocorato** [comment]
migrated from #747

File: purchase_product_weight/models/product.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Copyright 2019 Sergio Corato <https://github.com/sergiocorato>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import models, fields
```
Comment (Trivedi-Vacha-SerpentCS) [inline purchase_product_weight/models/product.py]: Import should be in alphabetical order

File: purchase_product_weight/models/product_supplierinfo.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Copyright 2019 Sergio Corato <https://github.com/sergiocorato>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import models, fields
```
Comment (Trivedi-Vacha-SerpentCS) [inline purchase_product_weight/models/product_supplierinfo.py]: Same as above

File: purchase_product_weight/models/purchase.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# Copyright 2019 Sergio Corato <https://github.com/sergiocorato>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import models, fields, api
```
Comment (Trivedi-Vacha-SerpentCS) [inline purchase_product_weight/models/purchase.py]: Same as above

File: purchase_product_weight/models/purchase.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    weight_total = fields.Float(string='Total weight')
```
Comment (Trivedi-Vacha-SerpentCS) [inline purchase_product_weight/models/purchase.py]: Good to define help on the field.

File: purchase_product_weight/models/product.py (hunk 1 above)
Comment (sergiocorato) [reply purchase_product_weight/models/product.py]: done

File: purchase_product_weight/models/purchase.py (hunk 3 above)
Comment (sergiocorato) [reply purchase_product_weight/models/purchase.py]: done

File: purchase_product_weight/models/product_supplierinfo.py (hunk 2 above)
Comment (sergiocorato) [reply purchase_product_weight/models/product_supplierinfo.py]: done, thanks

File: purchase_product_weight/models/purchase.py (hunk 4 above)
Comment (sergiocorato) [reply purchase_product_weight/models/purchase.py]: done

File: purchase_product_weight/models/purchase.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        help='Total weight computed from product weight * product quantity. '
+             'It can be written manually, in this case unit price will be '
+             'recomputed on Total weight / product quantity * seller price '
+             'if field "Compute price on weight" on product is flagged.')
+
+    @api.onchange('product_id', 'product_qty')
+    def _onchange_total_weight(self):
+        for line in self:
+            line.weight_total = line.product_id.weight * \
+                line.product_qty
```
Comment (ps-tubtim) [inline purchase_product_weight/models/purchase.py]: I think this function's name is `_onchange_weight_total` better than`_onchange_total_weight`. Do you think so?

Comment (sergiocorato) [reply purchase_product_weight/models/purchase.py]: Right, done

**Pull request closed without merging**